### DISCUSSION TITLE
Python code which cannot replace a string

### DISCUSSION THREAD

**Discussion by AbinReghunathan** [community/community#41908] [Other Feature Feedback, Questions, & Ideas]
req_data['Date']=req_data.Date.replace('<span class=','',inplace=True)
req_data['Date']

gives a None result

<html>
<body>

id | Reference_Name | Reference_Link | Date | Year
-- | -- | -- | -- | --
0 | General Python FAQ — Python 3.9.2 documentation | https://docs.python.org/3/faq/general.html#wha... | <span class="nowrap">28 March</span> | <span class="reference-accessdate">. Retrieved...
1 | Python 0.9.1 part 01/21 | https://www.tuhs.org/Usenet/alt.sources/1991-F... | <span class="nowrap">11 August</span> | <span class="reference-accessdate">. Retrieved...
2 | Python 3.11.1, 3.10.9, 3.9.16, 3.8.16, 3.7.16,... | https://pythoninsider.blogspot.com/2022/12/pyt... | <span class="nowrap">7 December</span> | <span class="reference-accessdate">. Retrieved...
3 | Why is Python a dynamic language and also a st... | https://wiki.python.org/moin/Why%20is%20Python... | <span class="nowrap">27 January</span> | <span class="reference-accessdate">. Retrieved...
4 | PEP 483 – The Theory of Type Hints | https://www.python.org/dev/peps/pep-0483/ | <span class="nowrap">14 June</span> | <span class="reference-accessdate">. Retrieved...

</body>
</html>

Unable to remove the span tag from the scraped data.

**Comment by VladislavDev** [comment]
What type of data does **req_data.Date** have? If you use **replace** for strings, it doesn't have a replace parameter. But it looks more like a **Pandas expression**. If so, you should remove **replace**. Also, you can not remove REPLACE, but leave only req_data.Date.replace('<span class=','',inplace=True).

**Comment by jge162** [comment]
remove the inplace=True argument from the replace()

```python
req_data['Date'] = req_data['Date'].replace('<span class=','')
print(req_data['Date']) 
```

**Reply by jge162** [reply]
similarly you can also try assign(). @AbinReghunathan 

```python
req_data = req_data.assign(Date = req_data['Date'].replace('<span class=',''))
print(req_data['Date'])
```
